lib/samples/style_graphics_with_renderer/style_graphics_with_renderer.dart

```
@@ -0,0 +1,276 @@
+    );
+    final parameters = GeodesicEllipseParameters(
+      axisDirection: -45,
+      angularUnit: AngularUnit(unitId: AngularUnitId.degrees),
+      center: ellipseCenter,
+      linearUnit: LinearUnit(unitId: LinearUnitId.kilometers),
+      maxPointCount: 100,
+      maxSegmentLength: 20,
+      geometryType: GeometryType.polygon,
+      semiAxis1Length: 200,
+      semiAxis2Length: 400,
+    );
+    final ellipseGeometry =
+        GeometryEngine.ellipseGeodesic(parameters: parameters);
+    final ellipseGraphic = Graphic(geometry: ellipseGeometry);
+
+    // Add the graphic to the overlay.
+    ellipseGraphicsOverlay.graphics.add(ellipseGraphic);
+    return ellipseGraphicsOverlay;
+  }
+
+  GraphicsOverlay getCurvedPolygonGraphicsOverlay() {
+    // Create a simple fill symbol with outline.
+    final curvedLineSymbol = SimpleLineSymbol(color: Colors.black);
+    final curvedFillSymbol = SimpleFillSymbol(
+      color: Colors.red,
+      outline: curvedLineSymbol,
+    );
+    // Create a graphics overlay for the polygons with curve segments.
+    final curvedGraphicsOverlay = GraphicsOverlay();
+    // Create and assign a simple renderer to the graphics overlay.
+    curvedGraphicsOverlay.renderer = SimpleRenderer(symbol: curvedFillSymbol);
+
+    // Create a heart-shape graphic from ArcGISSegment's.
+    final origin = ArcGISPoint(
+      x: 40e5,
+      y: 5e5,
+      spatialReference: SpatialReference.webMercator,
+    );
+    final heartGeometry = getHeartGeometry(center: origin, sideLength: 10e5);
+    final heartGraphic = Graphic(geometry: heartGeometry);
+
+    // Add the graphic to the overlay.
+    curvedGraphicsOverlay.graphics.add(heartGraphic);
+    return curvedGraphicsOverlay;
+  }
+
+  Geometry? getHeartGeometry({
+    required ArcGISPoint center,
+    required double sideLength,
+  }) {
+    // The side length should be always greater than 0.
+    if (sideLength <= 0) {
+      return null;
```

I think we should throw an ArgumentError rather than returning a null and silently ignoring it. Then the return type can be non-nullable.
```dart
      throw ArgumentError.value(sideLength, 'sideLength', 'Must be positive');
```

---

Yeah, that's a good design decision.